scripts/create_elmo_embeddings_from_vocab.py

```
@@ -0,0 +1,112 @@
+# pylint: disable=no-self-use
+
+import os
+import argparse
+import sys
+
+sys.path.insert(0, os.path.dirname(os.path.abspath(os.path.join(__file__, os.pardir))))
+
+import gzip
+import numpy
+import torch
+from torch.autograd import Variable
+
+from allennlp.data.token_indexers import ELMoTokenCharactersIndexer
+from allennlp.modules.elmo import _ElmoCharacterEncoder
+from allennlp.data import Token, Vocabulary
+
+
+def main(vocab_path: str,
+         elmo_config_path: str,
+         elmo_weights_path: str,
+         output_dir: str,
+         batch_size: int,
+         device: int):
+    """
+    Creates ELMo word representations from a vocabulary file.
+    ELMo requires 2 additional tokens: <S> and </S>. The first token
+    in this file is assumed to be an unknown token.
+
+    This script produces two artifacts: A new vocabulary file
+    with the <S> and </S> tokens inserted and a glove formatted embedding
+    file containing word : vector pairs, one per line, with all values
+    separated by a space.
+    """
+
+    # Load the vocabulary words and convert to char ids
+    with open(vocab_path, 'r') as vocab_file:
+        tokens = vocab_file.read().strip().split('\n')
+
+    # Insert the sentence boundary tokens which elmo uses at positions 1 and 2.
+    tokens = [tokens[0]] + ["<S>", "</S>"] + tokens[1:]
```

Can we throw an exception if `tokens[0] != '@@UNKNOWN@@'` ?